src/utils/cleaning_data.py

```diff
@@ -0,0 +1,278 @@
+        upper_threshold (float): The maximum percentage of indirect missing values allowed for columns to be included in the output. Default is None.
+
+    Returns:
+        list: A list of columns name in the filtered DataFrame
+    """
+    # Filter the Dataframe columns
+    data = df[columns_to_filter]
+    
+    # Get the maximum and minimum value of filtered data
+    max_value = data.max(axis=1)
+    min_value = data.min(axis=1)
+    
+    # Convert columns in the DataFrame to numeric data type
+    filtered_df = data.apply(pd.to_numeric, errors='coerce')
+
+    # Filter the DataFrame based on the user-provided thresholds for each column
+    if lower_threshold is None:
+        filtered_df = df.loc[max_value > upper_threshold]
+
+    elif upper_threshold is None:
+        filtered_df = df.loc[min_value < lower_threshold]
+    
+    else:
+        filtered_df = df.loc[
+            (max_value > lower_threshold) & 
+            (min_value < upper_threshold)
+        ]
+
+    # Return a list of the column names in the filtered DataFrame
+    return list(filtered_df.index)
+
+
+def is_one_hot_encoded(df: pd.DataFrame, col_name:str) -> bool:
+    """Check if the given column is one-hot encoded in the DataFrame.
+
+    Args:
+        df (pandas.DataFrame): The DataFrame to check.
+        col_name (str): The name of the column to check.
+
+    Returns:
+        bool: True if the column is one-hot encoded, False otherwise.
+    """
+    # Get unique values in the column
+    unique_vals = df[col_name].unique()
+    
+    # Create set of all possible values to check against
+    check_num = (0, 1, 999, 997, -9998, 995, -9999)
+
+    # Return True if the column is one-hot encoded
+    # by determine whether check_num is a subset of unique_vals
+    return set(check_num).issuperset(set(unique_vals))
+
+
+def _get_labeled_numeric_data(df: pd.DataFrame, col_names: list) -> tuple:
+    """Extracts labeled and numeric data from a DataFrame for a list of specified columns.
+
+    Args:
+        df (pd.DataFrame): A pandas DataFrame.
+        col_names (list): A list of column names to extract labeled and numeric data from.
+
+    Returns:
+        tuple: A tuple containing a list of the names of the numeric columns and a list of the names of the labeled columns.
+    """
+    # Initialize empty lists to store labeled and numeric columns
+    labeled_cols = []
+    numeric_cols = []
+    
+    # Loop through the specified column names
+    for col_name in col_names:
+        col = df[col_name]
+        
+        # Check if the column is numeric and not one-hot encoded
+        if pd.api.types.is_numeric_dtype(col) and not is_one_hot_encoded(df, col_name):
+            
+            # Check if the column is labeled
+            if col.is_monotonic_increasing or col.is_monotonic_decreasing or col.nunique() / col.size <= 0.05:
```

Comment:
Can u please describe how "is_monotonic_increasing" and "is_monotonic_decreasing" methods help in figuring out if the column is labeled or not

Reply:
The sorted attributes are often presented in ascending or descending order. We can identify between ordered and numeric attributes using this feature. For example, the degree of education attribute has a value/label rank ranging from 0 (uneducated) to 12. (a certain major degree). We shall differentiate that order in this capacity is an ordered/labeled attribute, as opposed to natural numbers having a clear meaning, such as a person's height.

Reply:
But because we have a missing error code in the data, it ruins the arrangement and return always false